lib/utils/pyexec.c

```diff
@@ -547,10 +547,12 @@ int pyexec_file_if_exists(const char *filename) {
         return pyexec_frozen_module(filename);
     }
     #endif
-    if (mp_import_stat(filename) != MP_IMPORT_STAT_FILE) {
-        return 1; // success (no file is the same as an empty file executing without fail)
+    #if MICROPY_VFS || MICROPY_MBFS
+    if (mp_import_stat(filename) == MP_IMPORT_STAT_FILE) {
```

Comment:
`mp_import_stat()` should always be provided by a port, regardless of the `MICROPY_VFS` setting.  In other words, this code is still valid if `MICROPY_VFS` is disabled, so shouldn't be excluded with this `#if`

Reply:
Ok. Removed this change.